contracts/BSCValidatorSet.sol

```diff
@@ -421,28 +491,28 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
 
   /*********************** Internal Functions **************************/
 
-  function checkValidatorSet(Validator[] memory validatorSet) private pure returns(bool, string memory) {
-    if (validatorSet.length > MAX_NUM_OF_VALIDATORS){
+  function checkValidatorSet(Validator[] memory validatorSet) private pure returns (bool, string memory) {
+    if (validatorSet.length > MAX_NUM_OF_VALIDATORS) {
       return (false, "the number of validators exceed the limit");
     }
-    for (uint i = 0;i<validatorSet.length;i++) {
-      for (uint j = 0;j<i;j++) {
+    for (uint i = 0; i < validatorSet.length; i++) {
+      for (uint j = 0; j < i; j++) {
         if (validatorSet[i].consensusAddress == validatorSet[j].consensusAddress) {
           return (false, "duplicate consensus address of validatorSet");
         }
       }
     }
-    return (true,"");
+    return (true, "");
   }
 
-  function doUpdateState(Validator[] memory validatorSet) private{
+  function doUpdateState(Validator[] memory validatorSet) private {
```

Comment:
We need clean  and re-init`validatorExtraSet` in this function

Reply:
We need operate `validatorExtraSet` when `currentValidatorSet.push` and `currentValidatorSet.pop` occur.

Reply:
Good catch